lib/auth.ts

```diff
@@ -1,11 +1,29 @@
-import { hash, compare } from 'bcryptjs';
+import { compare, hash } from "bcryptjs";
+import { DefaultSession } from "next-auth";
+import { getSession as getSessionInner, GetSessionOptions } from "next-auth/client";
 
-export async function hashPassword(password) {
-    const hashedPassword = await hash(password, 12);
-    return hashedPassword;
+export async function hashPassword(password: string) {
+  const hashedPassword = await hash(password, 12);
+  return hashedPassword;
 }
 
-export async function verifyPassword(password, hashedPassword) {
-    const isValid = await compare(password, hashedPassword);
-    return isValid;
-}
\ No newline at end of file
+export async function verifyPassword(password: string, hashedPassword: string) {
+  const isValid = await compare(password, hashedPassword);
+  return isValid;
+}
+
+type DefaultSessionUser = NonNullable<DefaultSession["user"]>;
+type CalendsoSessionUser = DefaultSessionUser & {
+  id: number;
+  username: string;
+};
+export interface CalendsoSession extends DefaultSession {
```

Comment:
```suggestion
interface Session extends DefaultSession {
```

@emrysal you'd prefer this?